monkestation/code/modules/projectiles/projectile/special/rocket.dm

```diff
@@ -0,0 +1,14 @@
+/obj/item/projectile/bullet/spatialriftnullifier
+	name = "Spatial Rift Nullifier Rocket"
+	icon_state = "missle"
+	damage = 10
+
+/obj/item/projectile/bullet/spatialriftnullifier/on_hit(atom/target, blocked = FALSE)
```

Comment:
Maybe make the target variable /obj/singularity just so it only catches that.
Then if there is no target, just detonate